app/src/main/java/com/example/taskwise_eventmaster/presentation/home_page/HomeScreen.kt

```diff
@@ -4,35 +4,35 @@ import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.rememberScrollState
 import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.SnackbarHostState
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import com.example.taskwise_eventmaster.presentation.sign_in.UserData
+import androidx.navigation.NavHostController
 
 
 @Composable
-fun HomePage(
-    userData: UserData?,
-    goToProfilePage: () -> Unit,
-    goToTaskPage: () -> Unit,
-    goToGoalsPage: () -> Unit,
-    goToEventsPage: () -> Unit,
-    goToPlanningViewPage: () -> Unit,
+fun HomeScreen(
+    state: HomeScreenState,
+    navController: NavHostController,
+    snackbarHostState: SnackbarHostState
 ) {
     Column(
         modifier = Modifier
             .fillMaxSize()
             .verticalScroll(rememberScrollState()),
         horizontalAlignment = Alignment.CenterHorizontally
     ) {
-        ProfilePart(userData = userData, goToProfilePage = goToProfilePage)
+        if (state.userData == null) {
+            //TODO print error message
```

Comment:
Not only print error message, but also handle the error properly. Most likely by showing a snackbar and navigating back or presenting a generic error screen with the option to relog.

Reply:
Could this be addressed in the context of this PR or should we leave it as such?